libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/binder/MapViewBinder.kt

```diff
@@ -19,49 +18,61 @@ import com.mapbox.navigation.dropin.component.marker.FreeDriveLongPressMapCompon
 import com.mapbox.navigation.dropin.component.marker.GeocodingComponent
 import com.mapbox.navigation.dropin.component.marker.MapMarkersComponent
 import com.mapbox.navigation.dropin.component.marker.RoutePreviewLongPressMapComponent
+import com.mapbox.navigation.dropin.coordinator.MapLayoutCoordinator
 import com.mapbox.navigation.dropin.databinding.MapboxNavigationViewLayoutBinding
 import com.mapbox.navigation.dropin.internal.extensions.reloadOnChange
-import com.mapbox.navigation.ui.app.internal.Store
 import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
 import com.mapbox.navigation.ui.base.lifecycle.UIBinder
 import com.mapbox.navigation.ui.maps.internal.ui.LocationComponent
 import com.mapbox.navigation.ui.maps.internal.ui.LocationPuckComponent
 import com.mapbox.navigation.ui.maps.internal.ui.RouteArrowComponent
 import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponent
-import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponentContract
 import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowOptions
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.combine
 
 @ExperimentalPreviewMapboxNavigationAPI
-@FlowPreview
-internal class MapBinder(
-    private val context: NavigationViewContext,
-    private val binding: MapboxNavigationViewLayoutBinding,
-    private val mapView: MapView
-) : UIBinder {
-
-    init {
-        mapView.compass.enabled = false
+abstract class MapViewBinder : UIBinder {
```

Comment:
Great initiative. I like the extraction of layout inflation logic from the MapLayoutCoordinator. 

However, I think we can improve this further by hiding map style reload logic within MapLayoutCoordinator and controlling it by introducing MapStyleLoadPolicy returned by MapViewBinder. 

Something like this:
![drop in - mapbinder refactor](https://user-images.githubusercontent.com/2678039/193631721-666e3040-805c-48f1-a567-c5007aecf551.png)
[diagram src](https://www.plantuml.com/plantuml/uml/dPJVJzim4CVV_LVSNWOy9FoGzbA525OLXiJAn0PFI9CJNrjFhbtPJakqxN_VsIwDqeL9p0zj7Dzz_FtUsRbmdblVB1KxM2ckmrDscf-Td9tb9_aFRXbxvnUuH5XowvYhkK8upJymLYlk77pcgrlUciQFZR52QkwDXPyCma9bdgTyBUVSIwFpcJARHucRZr8BjExmQ0HNzAf8yo8iNLlJhCgI_MPCQezsnck4XviK4QaLFHwkNq97iCiYslwETvL3kqORSAYRPGZuwbk5juQBBqR9ketCwUHnSa__TzFn9Fvz7zzDhswk7-ullzs4sQVBwVKaK7ZblEMrXvttpb145LKPXMkkYkEgXDhGxDcFe5CLCiRfzK22xiK5ijJpVMjfCPLz6qAeZ4OYcXLgCEJSIe41-tzb0yXf1AGiYVfYFXcpo3rk2_0FSfWMnxBiXCNqEVg_cq4dOgX5vs7_sACjw9n6u3L1bVCakevGWJFUA9_QGR1-VuhZ9XwirBfaUh21NU7Uw7CW_Sq-NpbyvBmJtqj6ozX1rX-CxUvCyIlF-tLWtIm5PDbGDTZWnSw4sUYit5CmWfbHT9NTWAqdpRK0HNC7BYp1Pe4QDA90yQGR5mxly5vMpXS-A_loH-5EpUIyiUYUDBm7YxwnsZ76jF48CGdCR41Ax4ywIR4v6GYZ3tmK5yAJFWPf7CPPN2I9i20R2WLKBIoCyy1NAsB62tEKmE7paRY9cFADLYtzr3jkjB-r1D815uAmtknz8YBpzUFGt-Rjy3H2Qon63AxHjk3b4a7WMdPwnzjonsIegGvpT0mfCzIRNT1JirH_0000)

Furthermore, we shouldn't make the `bind` method final. Opening it will allow subclasses to register addional components that must be managed alongside the MapView. (eg. [MapboxInfoPanelBinder](https://github.com/mapbox/mapbox-navigation-android/blob/24cb06d2059d2501bfc183a2b155703a697cbb65/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/binder/infopanel/InfoPanelBinder.kt#L132))